Library/Homebrew/extend/os/linux/extend/pathname.rb

```diff
@@ -6,8 +6,19 @@ def elf?
   end
 
   # @private
-  def dynamic?
-    @which_file ||= which("file")
-    !@which_file.nil? && `#{@which_file} -b #{self}` =~ /dynamic|shared/
+  def dynamic_elf?
+    if !which("readelf").nil?
+      needed = `readelf -d #{self} | grep NEEDED`
+      return false if needed.empty?
+    elsif !which("objdump").nil?
+      needed = `objdump -p #{self} | grep NEEDED`
+      return false if needed.empty?
```

Comment:
Since both `readelf` and `objdump` are provided by `binutils`, it's unlikely that the user would have one and not the other. I'd be inclined to get rid of `objdump`.

Reply:
> it's unlikely that the user would have one and not the other. I'd be inclined to get rid of `objdump`.

macOS users aside - yes. I included it because it works with more formats and is as safe as `readelf`.